src/js/jsx/sections/export/ExportAllPanel.jsx

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2014 Adobe Systems Incorporated. All rights reserved.
+ *  
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"), 
+ * to deal in the Software without restriction, including without limitation 
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, 
+ * and/or sell copies of the Software, and to permit persons to whom the 
+ * Software is furnished to do so, subject to the following conditions:
+ *  
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *  
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, 
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER 
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
+ * DEALINGS IN THE SOFTWARE.
+ * 
+ */
+
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var React = require("react"),
+        Fluxxor = require("fluxxor"),
+        FluxMixin = Fluxxor.FluxMixin(React),
+        StoreWatchMixin = Fluxxor.StoreWatchMixin,
+        classnames = require("classnames"),
+        _ = require("lodash");
+
+    var Button = require("jsx!js/jsx/shared/Button"),
+        CheckBox = require("jsx!js/jsx/shared/CheckBox"),
+        TitleHeader = require("jsx!js/jsx/shared/TitleHeader");
+
+    var strings = require("i18n!nls/strings"),
+        ExportAsset = require("js/models/exportasset");
+
+    /**
+     * Private React component that represents one Layer and its configured export assets
+     */
+    var PanelLayer = React.createClass({
+        mixins: [FluxMixin],
+
+        /**
+         * Handle the change of this layer's checkbox, update the exportEnabled flag accordingly
+         *
+         * @private
+         * @param {SyntheticEvent} event
+         * @param {boolean} checked 
+         */
+        _handleLayerSelectedChanged: function (event, checked) {
+            this.getFlux().actions.export.setLayerExportEnabled(this.props.document, this.props.layer, checked);
+        },
+
+        render: function () {
+            var layer = this.props.layer,
+                layerExports = this.props.layerExports,
+                freshState = this.props.freshState;
+
+            // The list of assets within a layer
+            var assetsComponent = layerExports.map(function (asset, key) {
```

Comment:
Well, we should stop even bothering with this discussion since the performance issue is so minimal, but: if you deleted the 0th asset, then I believe React would then do a DOM-level update to assets 1...N instead of just removing the 0th one from the DOM. This could be solved by giving each asset a counter-based ID.